### PR TITLE
Add scratch-link cask

### PR DESCRIPTION
**Pull request by bjrbhre**
version '1.3.57'

Scratch Link is a helper application which allows Scratch 3.0 to communicate with hardware peripherals.
Scratch Link replaces the Scratch Device Manager and Scratch Device Plug-in.
It is open-sourced (BSD) since 2019.

The source code is available here
https://github.com/LLK/scratch-link

To learn more about Scratch & Scratch Link go here
https://scratch.mit.edu/microbit

This is also where the zip/pkg download link is provided
https://downloads.scratch.mit.edu/link/mac.zip

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

**Comment by bjrbhre** [comment]
I've also opened a discussion on scratch-link side to be sure they approved their pkg being available through cask
https://github.com/LLK/scratch-link/issues/182

File: Casks/scratch-link.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+cask "scratch-link" do
+  version "1.3.57"
+  sha256 "58e9e3a5745bc65e6920bed934eb952e29b184022a9c6d8942dfcdf13b644c9d"
+
+  # downloads.scratch.mit.edu was verified as official when first introduced to the cask
+  url "https://downloads.scratch.mit.edu/link/mac.zip"
```
Comment (miccal) [inline Casks/scratch-link.rb:6]: How does one find this download link from the `homepage`?

File: Casks/scratch-link.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  name "Scratch Link"
+  homepage "https://github.com/LLK/scratch-link"
+
+  license :bsd
```
Comment (miccal) [inline Casks/scratch-link.rb:10]: Please remove the `licence` stanza -- this is no longer required in cask's.

File: Casks/scratch-link.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+cask "scratch-link" do
+  version "1.3.57"
+  sha256 "58e9e3a5745bc65e6920bed934eb952e29b184022a9c6d8942dfcdf13b644c9d"
+
+  # downloads.scratch.mit.edu was verified as official when first introduced to the cask
+  url "https://downloads.scratch.mit.edu/link/mac.zip"
+  name "Scratch Link"
+  homepage "https://github.com/LLK/scratch-link"
```
Comment (miccal) [inline Casks/scratch-link.rb:8]: Please change this to:
```
homepage "https://scratch.mit.edu/microbit"
```

File: Casks/scratch-link.rb (hunk 2 above)
Comment (bjrbhre) [reply Casks/scratch-link.rb]: Fixed by bjrbhre/homebrew-cask@a85720cc96cf9461d94652d7b53d349445b1dc8b

File: Casks/scratch-link.rb (hunk 3 above)
Comment (bjrbhre) [reply Casks/scratch-link.rb]: Fixed by bjrbhre/homebrew-cask@a85720cc96cf9461d94652d7b53d349445b1dc8b

**Comment by vitorgalvao** [comment]
Needs to be submitted to [homebrew/cask-drivers](https://github.com/Homebrew/homebrew-cask-drivers). We [define drivers](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) as:

> Software to make a hardware peripheral recognisable and usable by the system. If the software is useless without the peripheral, it’s considered a driver.

Note that repo has [special naming rules](https://github.com/Homebrew/homebrew-cask-drivers/blob/master/CONTRIBUTING.md#naming-driver-casks).

**Pull request closed without merging**